app.py

```diff
@@ -71,22 +139,55 @@ def render_and_submit_version(self, template, fields, first_frame, last_frame, s
         """
+
+        tk_multi_reviewsubmission = self.import_module("tk_multi_reviewsubmission")
+
+        # Is the app configured to do anything?
+        upload_to_shotgun = self.get_setting("upload_to_shotgun")
+        store_on_disk = self.get_setting("store_on_disk")
+        if not upload_to_shotgun and not store_on_disk:
+            self.log_warning("App is not configured to store images on disk nor upload to shotgun!")
+            return None
+
         # Make sure we don't overwrite the caller's fields
         fields = copy.copy(fields)
 
-        # Tweak fields so that we'll be getting nuke formatted sequence markers (%03d, %04d etc):
-        for key_name in [key.name for key in template.keys.values() if isinstance(key, sgtk.templatekey.SequenceKey)]:
-            fields[key_name] = "FORMAT: %d"
+        # Get the name for the Version entity
+        version_template = self.get_template("sg_version_name_template")
+        version_name = None
+        if version_template:
+            version_name = version_template.apply_fields(fields)
 
-        # Get our input path for frames to convert to movie
-        path = template.apply_fields(fields)
+        # Submit Version
+        progress_cb(50, "Creating Shotgun Version and uploading movie")
+        submitter = tk_multi_reviewsubmission.Submitter()
+        sg_version = submitter.submit_version(path,
+                                              path_to_upload,
+                                              thumbnail_path,
+                                              sg_publishes,
+                                              sg_task,
+                                              comment,
+                                              store_on_disk,
+                                              first_frame,
+                                              last_frame,
+                                              upload_to_shotgun,
+                                              version_name)
 
-        # call new version
-        return self.render_and_submit_path(path, fields, first_frame, last_frame, sg_publishes, sg_task,
-                                  comment, thumbnail_path, progress_cb, color_space, *args, **kwargs)
+        # Remove from filesystem if required
+        if not store_on_disk and os.path.exists(path_to_upload):
+            progress_cb(90, "Deleting rendered movie")
+            os.unlink(path_to_upload)
+
+        # log metrics for this app's usage
+        try:
+            self.log_metric("Render & Submit Version", log_version=True)
+        except:
+            # ignore any errors. ex: metrics logging not supported
+            pass
 
+        return sg_version
 
-    def render_and_submit_path(self, path, fields, first_frame, last_frame, sg_publishes, sg_task,
+    def render_and_submit_version(self, path, fields, first_frame, last_frame, sg_publishes, sg_task,
```

Comment:
Please do not change the signature of any previously existing public api methods. We need to maintain both `render_and_submit_version` as well as `render_and_submit_path` as-is.

Reply:
Please also change `path` to `path_to_frames` for `render_and_submit_path`